### PR TITLE
Issue #4281 Add performance tests for advanced pom support

### PR DESCRIPTION
**Pull request by ljacomet**
### Context
See #4281

### CODE REVIEW

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/LargeDependencyGraphPerformanceTest.groovy [hunk 1]
```diff
@@ -94,26 +86,46 @@ class LargeDependencyGraphPerformanceTest extends AbstractCrossVersionPerformanc
...
+        def result = runner.run()
+
+        then:
+        result.assertCurrentVersionHasNotRegressed()
+
+        cleanup:
+        stopServer()
+    }
+
+    def "resolve large dependency graph with advanced pom support (parallel)"() {
```
Comment (bigdaz) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/LargeDependencyGraphPerformanceTest.groovy]: I don't think we need this variant of the test. Our long term goal is to make parallelization something that 'just happens' and to deprecate the `--parallel` flag.

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/LargeDependencyGraphPerformanceTest.groovy [hunk 2]
```diff
@@ -94,26 +86,46 @@ class LargeDependencyGraphPerformanceTest extends AbstractCrossVersionPerformanc
...
-    private static BuildExperimentListener createArgsTweaker(String baseline) {
-        new BuildExperimentListener() {
-            @Override
-            void beforeExperiment(BuildExperimentSpec experimentSpec, File projectDir) {
-                GradleInvocationSpec invocation = experimentSpec.invocation as GradleInvocationSpec
-                if (invocation.gradleDistribution.version.version != baseline) {
-                    invocation.args << '-Dorg.gradle.advancedpomsupport=true'
-                }
-            }
+    def "resolve large dependency graph with advanced pom support"() {
```
Comment (bigdaz) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/LargeDependencyGraphPerformanceTest.groovy]: Now that we have several similar examples, we should consider using `where:`  and `@Unroll` to define the different test scenarios.

Comment (ljacomet) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/LargeDependencyGraphPerformanceTest.groovy]: Going to use this as an excuse to learn that feature

Comment (bigdaz) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/LargeDependencyGraphPerformanceTest.groovy]: Check out this example: https://github.com/gradle/gradle/blob/5a1a8781d8d0fd1c866d64d1e7bacffd4294da55/subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy#L33-L55

**Comment by jjohannes** [comment]
@ljacomet to verify that everything works on CI, you should run a full branch pipeline build (we do not do that automatically). The results will then show up here in the "checks" section.

- Go to https://builds.gradle.org/project.html?projectId=Gradle_Check&tab=projectOverview&branch_Gradle_Check
- select your branch
- start a "Branch Accept (Trigger)" build

**Comment by melix** [comment]
(and in "select your branch", just type in the number of the PR, that is to say 4291 here)

**Pull request merged**